### DISCUSSION TITLE
Active Record 6 Support

### DISCUSSION THREAD

**Discussion by rzane** [rzane/baby_squeel#110] [General]
The purpose of this discussion is to provide support to those who would see to add support for Active Record 6.

**Comment by marnen** [comment]
At a high level, what is standing in the way of Rails 6 support? Understanding that is the first key.

**Comment by rzane** [comment]
You'll have to run the tests to find out!

At one point, I had hoped that Ransack and BabySqueel could share the logic for converting a relation to a join dependency. That didn't end up working out, so I've merged that back into this project to make it more manageable. @vitalinfo submitted a PR there for AR 6 support, but I never really got around to checking it out: https://github.com/rzane/join_dependency/pull/6

Once that's one, you'll probably run into fun new slew of issues.

**Comment by rocket-turtle** [comment]
Hey @marnen,

I'm working on our Rails 5.2 upgrade and out of curiosity try to run the tests with activerecord 6.0.
(just changed activerecord and sqlite3 in baby_squeel/baby_squeel.gemspec and run bundle update)

I saw 82 failures. All of them where one of these two errors:

      ArgumentError:
        wrong number of arguments (given 3, expected 2)
      # ./lib/baby_squeel/active_record/query_methods.rb:48:in `build_joins'

      ArgumentError:
        wrong number of arguments (given 3, expected 4)
      # /Users/jonas.schoeler/.rvm/gems/ruby-2.6.6@baby/gems/activerecord-6.0.3.7/lib/active_record/associations/join_dependency.rb:67:in `initialize'
      # ./lib/baby_squeel/join_dependency.rb:125:in `new'

But befor starting to upgrade to a new Version I would suggest removing the old code for AR 4.
That might be a good way into the project. What do you thing?

I'm willing to help but as I said I'm on the AR 5.2 update so it might take a while till I'm looking at AR 6.

I would love to see this gem moving forward.

**Reply by marnen** [reply]
Thanks for the information! I don’t think that removing the code for Rails 4 is as high a priority as making the gem compatible with Rails 6, so I don’t think that’s a good beginning point. (I’m perfectly happy to break compatibility with Rails 4 if it’s necessary in order to easily make the gem be compatible with Rails 6, but I think doing so gratuitously is a waste of time.)

**Reply by rocket-turtle** [reply]
If @rzane is okay with it I can start / try to do the cleanup.
In my opinion an cleaner gem would be more easy to upgrade.

**Reply by rzane** [reply]
If there is still Active Record 4 stuff, feel free to remove it. I already removed support for it because Active Record 4.2 hasn't received security patches for quite some time.

**Reply by rocket-turtle** [reply]
Okay did some cleanups https://github.com/rzane/baby_squeel/pull/114/commits

**Reply by rocket-turtle** [reply]
If been working on the Rails 6.0 Support:
https://github.com/rocket-turtle/baby_squeel/tree/activerecord_6
https://github.com/rocket-turtle/baby_squeel/commit/14adeec5c20b26eeb500847f4a7470ce03287f09

All Tests are green :) but I have no Rails 6.0 application to test it in real live.
It is based on my other MR https://github.com/rzane/baby_squeel/pull/114/files so I will create a MR after this get merged.
